### DISCUSSION TITLE
Make lightweight for RaspberryPI(Deploy only necessary components)

### DISCUSSION THREAD

**Discussion by veerendra2** [immich-app/immich#1961] [Q&A] [answered]
Hello, I want to deploy Immich on RaspberryPI4, I see the [architecture](https://immich.app/docs/developer/architecture), there are components 
* `server`
* `microservice`
* `machine learning`
* DBs(`postgres` and `redis`). 

From [docker-compose.yml](https://github.com/immich-app/immich/blob/main/docker/docker-compose.yml), there are 
* `immich-server`
* `immich-microservices`
* `typesense`
* `immich-machine-learning`
* `immich-web`
* DBs(`postgres` and `redis`)
* `immich-proxy`

I want to make Immich lightweight i.e. minimize to necessary components. Which components are necessary so that Immich up and running. I guess below are necessary and proxy I want to use `traefik`? can you confirm?
* `immich-server`
* `immich-microservices`
* DBs

Any suggestions would be appreciated

Thanks

**Comment by bo0tzz** [comment, marked as answer]
The required components are
* immich-server
* immich-web
* immich-microservices
* immich-proxy (or recreate [the proxy configuration](https://github.com/immich-app/immich/blob/main/nginx/templates/default.conf.template) yourself if you're up for that)
* redis
* postgres

Machine-learning is optional, and can be disabled by removing it from the docker-compose.yml and setting `IMMICH_MACHINE_LEARNING_URL=false`.

Note that typesense isn't finished yet, and will be coming in the next release. You should use the docker-compose.yml that's attached to [the latest release](https://github.com/immich-app/immich/releases/tag/v1.50.1). We want to make typesense an optional component as well, but it might be required for a few releases until we get around to that. (It also isn't that resource heavy, so you should be OK there)

Do note that Immich is not designed to be a lightweight app. Disabling ML will definitely get you some savings, but during a large library import Immich will still use a lot of resources. Other people have run it okay on a raspi, but it will probably take some patience. It might be worth [limiting the resource usage](https://docs.docker.com/compose/compose-file/compose-file-v3/#resources) of the microservices container if it gets too hungry.

**Reply by veerendra2** [reply]
Thanks, Im trying to deploy Immich on docker swarm RPi cluster. I will change accordingly.

**Reply by raisinbear** [reply]
@bo0tzz Since I'm also on raspberry pi: Does the typesense requirement in the next few releases mean that arm/v7 will be dropped? The typesense docker images are lmited to amd64 and arm64/v8 after all..

**Reply by bo0tzz** [reply]
Typesense will probably be required at first, but we do intend to make it an optional component.

**Reply by raisinbear** [reply]
Ok thanks for the info, anyways! I will then stay away from server / mobile updates until compatibility is back.
